packages/ui/src/components/va-tree-view/components/VaTreeNode/VaTreeNode.vue

```diff
@@ -123,23 +128,28 @@ export default defineComponent({
 
     const tabIndexComputed = computed(() => props.node.disabled ? -1 : 0)
 
-    const handleToggleNode = (event: Event, node: TreeNode) => {
-      if (node.expanded) {
-        (event.target as HTMLElement)?.blur()
-      } else {
-        toggleNode(node)
+    const onNodeClick = (type: typeof expandNodeBy) => {
+      type = expandNodeBy === 'node' && type === 'leaf' ? 'node' : type
```

Comment:
I recommend creating new variable instead of reassigning arguments. The name of the variable will hint to what what code does making it more readable.